hack/periodic-mirroring.sh

```
@@ -23,13 +28,128 @@ else
+	while [ "${_page}" -le 50 ]; do
+		_url="https://quay.io/api/v1/repository/openshift/ci/tag/?onlyActiveTags=true&filter_tag_name=like:${_prefix}_&limit=100&page=${_page}"
+		_body="$(curl -sS -f --connect-timeout 10 --max-time 60 -K "${_curl_cfg}" "${_url}")" || { rm -f "${_out}" "${_curl_cfg}"; return 1; }
+		_has_additional="$(printf '%s' "${_body}" | grep -oE '"has_additional"[[:space:]]*:[[:space:]]*(true|false)' | head -1 | grep -oE 'true|false' || true)"
+		[ -n "${_has_additional}" ] || _has_additional=false
+		_names="$(printf '%s' "${_body}" | grep -oE '"name":[[:space:]]*"[^"]+"' | sed 's/.*"\([^"]*\)"$/\1/' || true)"
+		[ -n "${_names}" ] || break
+		while IFS= read -r _full; do
+			[ -n "${_full}" ] || continue
+			case "${_full}" in
+			"${_p}"*)
+				_suf="${_full#${_p}}"
```

Shellcheck (0.11.0)

[info] Expansions inside ${..} need to be quoted separately, otherwise they match as patterns.

(SC2295)